src/main/java/com/mentorship/vineservice/service/impl/VineServiceImpl.java

```diff
@@ -31,24 +32,30 @@ public class VineServiceImpl implements VineService {
     private final VineRepository vineRepository;
 
     public Long saveVine(VineDto vine) {
-        return vineRepository.save(vineMapper.vineDtoToVine(vine)).getId();
+        return vineRepository.save(vineMapper.mapVine(vine)).getId();
     }
 
     public VinesDto getVinesWithFilterAndPagination(VinesQueryParameters vinesQueryParameters) {
 
         Pageable pageable = PageRequest.of(vinesQueryParameters.getPage(), vinesQueryParameters.getSize());
 
-        Page<Vine>  vinePages = vineRepository.findAll(Specification.where(
-            equalsName(vinesQueryParameters.getName()))
+        Page<Vine> vinePages = vineRepository.findAll(Specification.where(
+                equalsName(vinesQueryParameters.getName()))
             .and(equalsSugar(vinesQueryParameters.getSugar()))
             .and(equalsColor(vinesQueryParameters.getColor()))
             .and(equalsGrape(vinesQueryParameters.getGrapeName()))
             .and(equalsYear(vinesQueryParameters.getYear())), pageable);
 
         return VinesDto.builder()
             .totalCount(vinePages.getTotalElements())
-            .vines(vineMapper.vineListToVineDtoList(vinePages.getContent()))
+            .vines(vineMapper.mapVinesDto(vinePages.getContent()))
             .build();
     }
 
+    @Override
+    public Vine getVineById(Long vineId) {
+        return vineRepository.findById(vineId)
+            .orElseThrow(() -> new NoSuchElementException(String.format("Vine with id %s not found", vineId)));
```

Comment:
Can we change this exception to a custom? 
Pls look through this article https://www.baeldung.com/spring-response-status-exception fbout http code exceptions in spring